File: isSubsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 * https://leetcode.com/problems/is-subsequence/discuss/87302/Binary-search-solution-for-follow-up-with-detailed-comments
 */
var isSubsequence = function (s, t) {
  let si = 0;
  let ti = 0;
  while (ti < t.length) {
    if (t[ti++] === s[si]) {
      si++;
    }
  }
  return si === s.length;
};
isSubsequence('abc', 'ahbgdc');

File: leetcode/isSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        ns = len(s)
        nt = len(t)
        i = j = 0
        while i < ns and j < nt:
            if s[i] == t[j]:
                i += 1
                j += 1
            else:
                j += 1
        return i == ns
